### PR TITLE
feat(eventing): add switch over and nvme path events

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added switch over and nvme path events.

### CODE REVIEW

File: control-plane/agents/src/bin/ha/cluster/switchover.rs [hunk 1]
```diff
@@ -647,6 +663,32 @@ impl From<Operation> for Stage {
...
+        let status = match self.stage() {
+            Stage::Init => SwitchOverStatus::SwitchOverStarted,
+            Stage::Successful => SwitchOverStatus::SwitchOverCompleted,
+            Stage::Errored => SwitchOverStatus::SwitchOverFailed,
+            _ => SwitchOverStatus::UnknownSwitchOverStatus,
+        };
+
+        let event_source = EventSource::new(node_id.to_string()).with_switch_over_data(
+            status,
+            self.timestamp,
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: Is this when the path down message is received from agent-ha-node?

**Review by tiagolobocastro — APPROVED** [review]
Are we adding agent-ha-node events?

File: control-plane/agents/src/bin/ha/cluster/switchover.rs (hunk 1 above)
Comment (datacore-vvarakantham) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: Yes it is.

No I was adding just the agent-ha-cluster events. What agent-ha-node events could be added?

Comment (tiagolobocastro) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: In Node agent there are things we could pick up on, path suspected, fail or "fixed". These would be very useful to have.

Comment (datacore-vvarakantham) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: I added nvme path suspect and fail events now.

Comment (tiagolobocastro) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: Missing one for when the path is "fixed" ?

Comment (datacore-vvarakantham) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: Can we get the fixed from node agent? Can we differentiate between the fixed ones and the new ones when a new path is added?

Comment (datacore-vvarakantham) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs:678]: Added NvmePathConnect event.

**Review by chriswldenyer — APPROVED** [review]
Looks reasonable

**Comment by datacore-vvarakantham** [comment]
bors merge

**Pull request merged**